### PR TITLE
Peer timer, unblock command, good fixes.

### LINKED ISSUE #34: Timer for connected users
Add timer to WG (maybe watchdog service?) for connected users, and ability to set maximum usage time. We may need to add special table in the DB.

### PR DESCRIPTION
**Pull request by CreativeeBlackWolf**
Closes #34

### CODE REVIEW

**Comment by CreativeeBlackWolf** [comment]
ability to disable peers actually gives us the ability to terminate connections

File: config_example.conf [hunk 1]
```diff
@@ -5,6 +5,9 @@ admins = <admin,ids>
 [db]
 path=db.sqlite
 
+[core]
+peer_active_time=<hours>
```
Comment (head47) [inline config_example.conf:9]: perhaps specify a default amount of hours instead of a placeholder?

File: core/watchdog/events.py [hunk 2]
```diff
@@ -33,7 +42,19 @@ def __init__(self,
...
         during client connection checks"""
 
-    async def __check_connection(self, client: Client, peer: ConnectionPeer) -> bool:
+    async def __check_connection(self, client: Client, peer: ConnectionPeer, warn: bool = False) -> bool:
+        if isinstance(peer.peer_timer, datetime.datetime) and peer.peer_status == PeerStatusChoices.STATUS_CONNECTED:
+            timedelta = peer.peer_timer - datetime.datetime.now()
+
+            if timedelta <= datetime.timedelta(minutes=2):
+                # True is disable
+                await self.timer_observer.trigger(client, peer, True)
```
Comment (head47) [inline core/watchdog/events.py:51]: change "True" to "disconnect=True" to clarify what this bool stands for

File: core/watchdog/events.py [hunk 3]
```diff
@@ -33,7 +42,19 @@ def __init__(self,
...
+                await self.emit_timeout_disconnect(client, peer)
+                return False
+            elif timedelta <= datetime.timedelta(minutes=15) and warn:
+                # False is warning
+                await self.timer_observer.trigger(client, peer, False)
```
Comment (head47) [inline core/watchdog/events.py:56]: as above

File: core/watchdog/events.py [hunk 4]
```diff
@@ -33,7 +42,19 @@ def __init__(self,
         """Internal lock that prevents updating `self.clients`
         during client connection checks"""
 
-    async def __check_connection(self, client: Client, peer: ConnectionPeer) -> bool:
+    async def __check_connection(self, client: Client, peer: ConnectionPeer, warn: bool = False) -> bool:
+        if isinstance(peer.peer_timer, datetime.datetime) and peer.peer_status == PeerStatusChoices.STATUS_CONNECTED:
+            timedelta = peer.peer_timer - datetime.datetime.now()
+
+            if timedelta <= datetime.timedelta(minutes=2):
```
Comment (head47) [inline core/watchdog/events.py:49]: why 2 min earlier?

File: core/db/db_works.pyi [hunk 5]
```diff
@@ -25,7 +26,8 @@ class Client(BaseModel):
     def set_ip_address(self, ip_address: str) -> bool: ...
     def set_status(self, status: ClientStatusChoices) -> bool: ...
     def change_peer_name(self, peer_id: int, peer_name: str) -> bool: ...
-    def set_peer_status(self, peer_id: int, peer_status: PeerStatusChoices) -> bool: ...
+    def set_peer_status(self, peer_id: int, peer_status: PeerStatusChoices): ...
```
Comment (head47) [inline core/db/db_works.pyi]: specify that these functions return None

**Review by head47 — CHANGES_REQUESTED** [review]

File: config_example.conf (hunk 1 above)
Comment (CreativeeBlackWolf) [reply config_example.conf]: this is an example, no need to specify some amount

Comment (head47) [reply config_example.conf]: examples are usually modified *as needed*, that's the reason you wrote `path=db.sqlite` and not `path=<db_path>`

**Review by head47 — CHANGES_REQUESTED** [review]
https://github.com/CreativeeBlackWolf/heavens-gate/pull/43#discussion_r1826861790

**Review by head47 — APPROVED** [review]

**Pull request merged**